cpp/src/transform/transform.cpp

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/cudf.h>
+#include <utilities/cudf_utils.h>
+#include <cudf/transform.hpp>
+#include <cudf/column/column.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <cudf/utilities/traits.hpp>
+
+#include <utilities/column_utils.hpp>
+
+#include <cudf/legacy/column.hpp>
+
+#include <jit/launcher.h>
+#include <jit/type.h>
+#include <jit/parser.h>
+#include "jit/code/code.h"
+
+#include <types.h.jit>
+#include <types.hpp.jit>
+
+namespace cudf {
+namespace transformation {
+
+namespace jit {
+
+namespace detail {
+
+/**
+ * @brief Functor to enable the internal working of `get_data_ptr`
+ * @ref get_data_ptr
+ */
+template <typename column_view_type>
+struct get_data_ptr_functor {
+  template <typename T>
+  std::enable_if_t<is_fixed_width<T>(), const void *>
+  operator()(column_view_type& view) {
+    return static_cast<const void*>(view.template data<T>());
+  }
+  template <typename T>
+  std::enable_if_t<not is_fixed_width<T>(), const void *>
+  operator()(column_view_type& view) {
+    CUDF_FAIL("Invalid data type for transform operation");
+  }
+};
+
+/**
+ * @brief Get the raw pointer to data in a (mutable_)column_view
+ */
+template <typename column_view_type>
+auto get_data_ptr(column_view_type& view) {
```

Comment:
Since `mutable_column_view` is convertible to `column_view`, there's in usually no need to template like this unless you desire different behavior between `column_view` and `mutable_column_view`, but in this case it seems like a `const void*` is being returned, so I think a non-template should be sufficient. 
```suggestion
auto get_data_ptr(column_view const& view) {
```